doc/examples/tracking_tissue_classifier.py

```diff
@@ -0,0 +1,184 @@
+because it exit the image boundary or because the direction getter has no
+direction to follow. Each tissue classifier determines if the stoping is 'valid'
+or 'invalid'. A streamline is 'valid' when the tissue classifier determines the
+streamline stops in a position classified as 'ENDPOINT' or 'OUTSIDEIMAGE'. A
+streamline is 'invalid' when it stops in a position classified as 'TRACKPOINT'
+or 'INVALIDPOINT'. These conditions are described below. The 'LocalTracking'
+generator can be set to output all generated streamlines or only the 'valid'
+ones.
+
+This example is an extension of the
+:ref:``example_probabilistic_fiber_tracking`` example. We begin by loading the
+data and fitting a constrained spherical deconvolution (CSD) reconstruction
+model.
+"""
+import numpy as np
+
+from dipy.data import read_stanford_labels
+from dipy.reconst.csdeconv import ConstrainedSphericalDeconvModel
+from dipy.tracking import utils
+
+hardi_img, gtab, labels_img = read_stanford_labels()
+data = hardi_img.get_data()
+labels = labels_img.get_data()
+affine = hardi_img.get_affine()
+
+seed_mask = labels == 2
+white_matter = (labels == 1) | (labels == 2)
+seeds = utils.seeds_from_mask(seed_mask, density=1, affine=affine)
+
+csd_model = ConstrainedSphericalDeconvModel(gtab, None, sh_order=6)
+csd_fit = csd_model.fit(data, mask=white_matter)
+
+"""
+-- ThresholdTissueClassifier --
+A scalar map can be used to build a tissue classifier by thresholding it to a
+fixed value. Here, we show an example using the fractional anisotropy (FA) map
+of the DTI model. The threshold tissue classifier uses a trilinear
+interpolation method at the tracking position.
+
+Parameters:
+    metric_map: numpy array [:,:,:],
+    threshold: float.
+
+'ENDPOINT': metric_map < threshold,
+'OUTSIDEIMAGE': tracking point outside of metric_map,
+'TRACKPOINT': stopped because no direction are available,
+'INVALIDPOINT': N/A.
+"""
+
+import dipy.reconst.dti as dti
+from dipy.reconst.dti import fractional_anisotropy
+from dipy.tracking.local import ThresholdTissueClassifier
+
+tensor_model = dti.TensorModel(gtab)
+tenfit = tensor_model.fit(data, mask=white_matter)
+FA = fractional_anisotropy(tenfit.evals)
+
+threshold_classifier = ThresholdTissueClassifier(FA, .2)
+
+"""
+-- BinaryTissueClassifier --
+A binary mask can be used as tissue classifier for the tractography.
+Here, we show how to obtain the binary tissue classifier from
+the white matter mask defined above. The binary tissue classifier uses a
+nearest-neighbourhood interpolation method at the tracking position.
+
+Parameters:
+    mask: numpy array [:,:,:],
+
+'ENDPOINT': mask = 0,
+'OUTSIDEIMAGE': tracking point outside of mask,
+'TRACKPOINT': stopped because no direction are available,
+'INVALIDPOINT': N/A.
+"""
+from dipy.tracking.local import BinaryTissueClassifier
+
+binary_classifier = BinaryTissueClassifier(white_matter)
+
+"""
+-- ActTissueClassifier --
+Anatomically-constrained tractography (ACT) uses information from
```

Comment:
Might be worth referring to the Smith et al. paper here as well, and explain in a bit of detail what is and what isn't implemented from their ideas.